contrib/generic_proxy/filters/network/source/proxy.cc

```diff
@@ -371,11 +381,18 @@ void Filter::onDecodingFailure() {
   closeDownstreamConnection();
 }
 
-void Filter::writeToConnection(Buffer::Instance& data) {
+void Filter::writeToConnection(Buffer::Instance& buffer) {
   if (downstream_connection_closed_) {
     return;
   }
-  connection().write(data, false);
+  connection().write(buffer, false);
+}
+
+OptRef<Network::Connection> Filter::optionalConnection() {
+  if (downstream_connection_closed_) {
+    return {};
+  }
+  return {connection()};
```

Comment:
Do we really need the `optional` prefix the method name? this `connect()` seems like just a helper method, we can rename it as other name

Reply:
sgtm.

Reply:
done.